### ISSUE TITLE
Accessibility: The avatar and other menu controls don't indicate to users that they are menus.

### ISSUE THREAD

**Issue by MarcoZehe** [mozilla/discourse#234]
There are several links or buttons throughout Discourse that open menu-like popups. The avatar, notifications, and possibly others that I haven't found yet. All of these should indicate to assistive technologies that they have menus attached and that popups will open once pressed/clicked.

The general menu design pattern is [here](https://www.w3.org/TR/wai-aria-practices-1.1/#menubutton). Note that for some things, the [Disclosure pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure) might be more appropriate.

All of these don't announce to the user if their panels/popups were opened, and that something can be done. A very prominent example is the "Share a link to this topic" button. The panel to share the link, copy it etc., appears at the end of the DOM, but focus doen't move there, and no other indication is given that something actually happened after the button was pressed.

Note that the avatar image link also is lacking alternative text. Incidentally, the name inferred from the src attribute often gives the correct user name, however it would still be better if, while fixing the menu behavior for it, the alternative text would also be set appropriately to indicate that this is the user avatar.

Use the Accessibility Inspector to verify that things are set correctly, e. g. the names are something useful, the states have the has-popup in them, etc.

┆Issue is synchronized with this [Jira Bug](https://jira.mozilla.com/browse/DISC-2)
